src/styles/blocks/calendar.scss

```diff
@@ -1,17 +1,38 @@
 .calendar {
-  width: 706px;
+  width: $size*7 + $padding*2 + $gap*6;
+  padding: 10px;
   display: flex;
   flex-flow: row wrap;
   gap: 1px;
 
   &__day {
     @extend %cell-style;
     transition: 0.5s;
+    cursor: pointer;
 
     &:hover {
       transform: translateY(-10px);
-      cursor: pointer;
       background-color: pink;
     }
   }
 }
+
+@each $day, $gap in $days {
+  .calendar__day--#{$day} >
+  .calendar__day:first-child {
+    margin-left: #{$gap};
+  }
+}
+
+@for $count from 28 to 32 {
+  .calendar__length--#{$count} >
+  .calendar__day:nth-child(n+#{$count+1}) {
+    display: none;
+  }
+}
```

Comment:
You could also nest it to `&__day`, this way you should'n write `.calendar__day` and left only only `:nth-child` in line 29.

In line 28, think which way you could change `.calendar__length--` to `.calendar__day:` modifier(rename in html also). To nest it to  `.calendar__day`, because now it's an calendar block modifier and in nesting it wouldn't work properly.

More details in  [this video](https://mate.academy/learn/css-advanced/sass#/video/925) 2:44 timing